agent/pkg/proto/agent.proto

```diff
@@ -0,0 +1,272 @@
+syntax = "proto3";
+
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/pulumi/pulumi-kubernetes-operator/agent/pkg/proto";
+
+package agent;
+
+service AutomationService {
+
+  /**
+   * WhoAmI returns detailed information about the currently logged-in Pulumi
+   * identity.
+   */
+  rpc WhoAmI(WhoAmIRequest) returns (WhoAmIResult) {}
```

Comment:
Let's call these responses instead of results -- I think it's more idiomatic for proto.
```suggestion
  rpc WhoAmI(WhoAmIRequest) returns (WhoAmIResponse) {}
```